library/x509write_crt.c

```diff
@@ -537,6 +537,12 @@ int mbedtls_x509write_crt_pem( mbedtls_x509write_cert *crt,
         return( ret );
     }
 
+    if( ( size - olen ) >  0 )
```

Comment:
Why do this check explicitly? `memset` will likely perform it again (unless the compiler optimizes it, in which case it's clutter).